### PR TITLE
Cities: add a log for incomplete cities shape

### PR DESCRIPTION
**Pull request by antoine-de**
some cities shape are not correctly closed.

When it's the case we look for a node very closed to the non-closed 'end' node to
check if this look like an OSM problem.

http://ra.osmsurround.org/index can be used to double check the admin

### CODE REVIEW

File: source/cities/cities.cpp [hunk 1]
```diff
@@ -292,9 +292,23 @@ void OSMRelation::build_polygon(OSMCache& cache, std::set<u_int64_t> explored_id
...
+            next_node = next_way.nodes.back();
         }
         if (tmp_polygon.outer().size() < 2 || ref == references.end()) {
+            // add some logs
+            // some admins are at the boundary of the osm data, so their own boundary may be incomplete
+            // some other admins have a split boundary and it is impossible to compute it.
+            // to check if the boundary is likely to be split we look for a very near point
+            // The admin can also be checked with: http://ra.osmsurround.org/index
+            auto log = log4cplus::Logger::getInstance("log");
+            for (const auto& node: cache.nodes) {
```
Comment (TeXitoi) [inline source/cities/cities.cpp]: Noooooooooooooooo!

File: source/ed/osm2ed.cpp [hunk 2]
```diff
@@ -576,6 +580,20 @@ void OSMRelation::build_polygon(OSMCache& cache, std::set<u_int64_t> explored_id
             next_node = next_way->nodes.back();
         }
         if (tmp_polygon.outer().size() < 2 || ref == references.end()) {
+            // add some logs
+            // some admins are at the boundary of the osm data, so their own boundary may be incomplete
+            // some other admins have a split boundary and it is impossible to compute it.
+            // to check if the boundary is likely to be split we look for a very near point
+            // The admin can also be checked with: http://ra.osmsurround.org/index
+            auto log = log4cplus::Logger::getInstance("log");
+            for (const auto& node: cache.nodes) {
```
Comment (TeXitoi) [inline source/ed/osm2ed.cpp]: Againnnnnnnnnnnnnnn! ;-)

File: source/ed/osm2ed.cpp [hunk 3]
```diff
@@ -576,6 +580,20 @@ void OSMRelation::build_polygon(OSMCache& cache, std::set<u_int64_t> explored_id
...
+                if (node.osm_id != next_node->osm_id && node.almost_equal(*next_node)) {
+                    LOG4CPLUS_WARN(log, "Impossible to close the boundary of the admin " << name << " (osmid= "
+                                   << osm_id << "). The end node " << node.osm_id << " is almost the same as "
+                                   << next_node->osm_id << " it's likely that they are wrong dupplicate");
```
Comment (TeXitoi) [inline source/ed/osm2ed.cpp]: duPlicate

File: source/cities/cities.cpp [hunk 4]
```diff
@@ -292,9 +292,23 @@ void OSMRelation::build_polygon(OSMCache& cache, std::set<u_int64_t> explored_id
...
+                if (node.first != next_node->first && node.second.almost_equal(next_node->second)) {
+                    LOG4CPLUS_WARN(log, "Impossible to close the boundary of the admin " << name << " (osmid= "
+                                   << osm_id << "). The end node " << node.first << " is almost the same as "
+                                   << next_node->first << " it's likely that they are wrong dupplicate");
```
Comment (TeXitoi) [inline source/cities/cities.cpp]: duPlicate

**Comment by Hoshiyo** [comment]
Jenkins, test this please

**Pull request merged**